cccatalog-api/cccatalog/api/utils/watermark.py

```diff
@@ -1,15 +1,169 @@
+    if height < BREAKPOINT_DIMENSION:
+        smaller_dimension = smaller_dimension | Dimension.HEIGHT
+    return smaller_dimension
+
+
+def _get_font_path(monospace=False):
+    """
+    Return the path to the TTF font file
+    :param monospace: True for monospaced font, False for variable-width font
+    :return: the path to the TTF font file
+    """
+
+    font_name = 'SourceSansPro-Bold.ttf' \
+        if monospace \
+        else 'SourceSansPro-Bold.ttf'
+    font_path = os.path.join(
+        os.path.dirname(__file__),
+        'fonts',
+        font_name
+    )
+
+    return font_path
+
+
+def _fit_in_width(text, font, max_width):
+    """
+    Break the given text so that it fits in the given space
+    :param text: the text to fit in the limited width
+    :param font: the font containing size and other info
+    :param max_width: the maximum width the text is allowed to take
+    :return: the fitted text
+    """
+
+    char_width, _ = font.getsize('x')  # x has the closest to average width
+    max_chars = max_width // char_width
+
+    text = '\n'.join([
+        '\n'.join(wrap(line, max_chars))
+        for line in text.split('\n')
+    ])
 
+    return text
+
+
+# Framing
+
+def _create_frame(dimensions):
+    """
+    Creates an frame with the given dimensions
+    :param dimensions: a tuple containing the width and height of the frame
+    :return: a white frame with the given dimensions
+    """
+
+    return Image.new(
+        'RGB',
+        dimensions,
+        FRAME_COLOR
+    )
+
+
+def _frame_image(image, frame, left_margin, top_margin):
+    """
+    Fix the image in the frame with the specified spacing
+    :param image: the image to frame
+    :param frame: the frame in which to fit the image
+    :param left_margin: the margin to the left of the image
+    :param top_margin: the margin to the top of the image
+    :return: the framed image
+    """
+
+    frame.paste(image, (left_margin, top_margin))
+    return frame
+
+
+# Attribution
+
+
+def _full_license(image_info):
+    """
+    Get the full license from the image info
+    :param image_info: the information about a particular image
+    :return: the full license text for the image
+    """
+
+    license_name = image_info['license'].upper()
+    license_version = image_info['license_version'].upper()
+
+    return '{prefix}{name} {version}'.format(
+        prefix='' if license_name == 'CC0' else 'CC ',
+        name=license_name,
+        version=license_version
+    )
+
+
+def _get_attribution_text(image_info):
+    """
+    Generate the attribution text from the image info
+    :param image_info: the info pertaining to the licensing of the image
+    :return: the attribution text
+    """
+
+    title = image_info['title']
+    creator = image_info['creator']
+    full_license = _full_license(image_info)
+
+    return (
+        '{title}\n'
+        'By: {creator}\n'
+        'Licensed under: {full_license}'
```

Comment:
this should change to be all in one line, with the format `{title} by {creator} is licensed under {full_license}`. No `\n` anymore

Reply:
I'll make the change. Thanks :smiley: